README-ReactiveSwift.md

```diff
@@ -0,0 +1,355 @@
+    .throttle(0.5, onScheduler: QueueScheduler.mainQueueScheduler)
+    .on(event: { print ($0) }) // the side effect
+```
+
+This will print the stream's [events][Events], while preserving the original stream behaviour. Both [`SignalProducer`][Signal producers]
+and [`Signal`][Signals] provide the `logEvents` operator, that will do this automatically for you:
+
+```swift
+let searchString = textField.rac_textSignal()
+    .toSignalProducer()
+    .map { text in text as! String }
+    .throttle(0.5, onScheduler: QueueScheduler.mainQueueScheduler)
+    .logEvents()
+```
+
+For more information and advance usage, check the [Debugging Techniques](Documentation/DebuggingTechniques.md) document.
+
+## How does ReactiveSwift relate to Rx?
+
+ReactiveCocoa was originally inspired, and therefore heavily influenced, by
+Microsoft’s [Reactive
+Extensions](https://msdn.microsoft.com/en-us/data/gg577609.aspx) (Rx) library. There are many ports of Rx, including [RxSwift](https://github.com/ReactiveX/RxSwift), but ReactiveCocoa is _intentionally_ not a direct port.
+
+**Where ReactiveSwift differs from Rx**, it is usually to:
+
+ * Create a simpler API
+ * Address common sources of confusion
+ * More closely match Cocoa conventions
+
+The following are some of the concrete differences, along with their rationales.
+
+### Naming
+
+In most versions of Rx, Streams over time are known as `Observable`s, which
+parallels the `Enumerable` type in .NET. Additionally, most operations in Rx.NET
+borrow names from [LINQ](https://msdn.microsoft.com/en-us/library/bb397926.aspx),
+which uses terms reminiscent of relational databases, like `Select` and `Where`.
+
+**RAC is focused on matching Swift naming first and foremost**, with terms like
+`map` and `filter` instead. Other naming differences are typically inspired by
+significantly better alternatives from [Haskell](https://www.haskell.org) or
+[Elm](http://elm-lang.org) (which is the primary source for the “signal”
+terminology).
+
+### Signals and Signal Producers (“hot” and “cold” observables)
+
+One of the most confusing aspects of Rx is that of [“hot”, “cold”, and “warm”
+observables](http://www.introtorx.com/content/v1.0.10621.0/14_HotAndColdObservables.html) (event streams).
+
+In short, given just a method or function declaration like this, in C#:
+
+```csharp
+IObservable<string> Search(string query)
+```
+
+… it is **impossible to tell** whether subscribing to (observing) that
+`IObservable` will involve side effects. If it _does_ involve side effects, it’s
+also impossible to tell whether _each subscription_ has a side effect, or if only
+the first one does.
+
+This example is contrived, but it demonstrates **a real, pervasive problem**
+that makes it extremely hard to understand Rx code (and pre-3.0 ReactiveCocoa
+code) at a glance.
+
+[ReactiveCocoa 3.0][https://github.com/ReactiveCocoa/ReactiveCocoa/blob/master/CHANGELOG.md] has solved this problem by distinguishing side
```

Comment:
I thought it made sense, but this could probably use more cleanup down the road.